### PR TITLE
tests: run conformance test suite in Travis

### PR DESCRIPTION
**Pull request by lsm5**
Follow up to #1003

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

@TomSweeneyRedHat @nalind @rhatdan @mrunalp PTAL.
This currently takes < 15 mins on my fork.

### CODE REVIEW

**Comment by lsm5** [comment]
no idea whatsup, travis simply refuses to start on this.

**Comment by lsm5** [comment]
Travis seems busted for buildah. Also happening on https://github.com/containers/buildah/pull/1031

**Comment by lsm5** [comment]
but it's working for my fork o_O

File: hack/spc_build.sh [hunk 1]
```diff
@@ -21,11 +21,10 @@ dnf -y update --best --allowerasing
 dnf -y install dnf-plugins-core
 dnf -y copr enable lsm5/container-diff
 dnf -y install autoconf automake btrfs-progs-devel \
-   buildah \
-   container-diff device-mapper-devel golang go-md2man \
+   buildah container-diff device-mapper-devel golang go-md2man \
    git glibc-static gpgme-devel hostname iproute \
    iputils libassuan-devel libseccomp-static make \
-   moby-engine openssl
+   moby-engine podman
```
Comment (TomSweeneyRedHat) [inline hack/spc_build.sh]: Did you mean to remove openssl?

File: hack/spc_build.sh [hunk 2]
```diff
@@ -37,24 +36,10 @@ make GOPATH=$(pwd)/_build:$(pwd) all TAGS="seccomp containers_image_ostree_stub"
...
-echo "buildah push buildah-test-image..."
-./buildah push --cert-dir /home/travis/auth --tls-verify=false --authfile /root/.docker/config.json buildah-test-image localhost:5000/buildah-test-image
-
-echo "docker pull buildah-test-image..."
-docker pull localhost:5000/buildah-test-image
-
-echo "Running container-diff..."
-container-diff diff --type=rpm daemon://localhost:5000/buildah-test-image daemon://docker-test-image
+echo "Running conformance tests..."
+GOPATH=$(pwd) go test -v ./tests/conformance
```
Comment (TomSweeneyRedHat) [inline hack/spc_build.sh]: Are the removed *test-image tests taken care of elsewhere?

**Comment by lsm5** [comment]
On Thu, Sep 20, 2018 at 10:09:09AM -0700, Tom Sweeney wrote:
> TomSweeneyRedHat commented on this pull request.
> 
> 
> 
> > -echo "buildah bud dockerfile..."
> -./buildah --registries-conf tests/registries.conf bud --file hack/Dockerfile -t buildah-test-image .
> -
> -echo "buildah tag buildah-test-image..."
> -./buildah tag buildah-test-image localhost:5000/buildah-test-image
> -
> -echo "buildah push buildah-test-image..."
> -./buildah push --cert-dir /home/travis/auth --tls-verify=false --authfile /root/.docker/config.json buildah-test-image localhost:5000/buildah-test-image
> -
> -echo "docker pull buildah-test-image..."
> -docker pull localhost:5000/buildah-test-image
> -
> -echo "Running container-diff..."
> -container-diff diff --type=rpm daemon://localhost:5000/buildah-test-image daemon://docker-test-image
> +echo "Running conformance tests..."
> +GOPATH=$(pwd) go test -v ./tests/conformance
> 
> Are the removed *test-image tests taken care of elsewhere?

So, I was under the impression that tests/conformance handled all those cases
already, that's why the deletion of these lines. Was that a wrong assumption?


-- 
Lokesh
IRC: lsm5
GPG: 0xC7C3A0DD
https://keybase.io/lsm5

File: hack/spc_build.sh (hunk 1 above)
Comment (lsm5) [reply hack/spc_build.sh]: yup, openssl was only added earlier cause I was trying to move registry config inside fedora container.

Comment (lsm5) [reply hack/spc_build.sh]: ohh btw, just to be clear, openssl installed by default on the base fedora container, so it's just that I'm just reverting to not installing it here in the first place.

Comment (TomSweeneyRedHat) [reply hack/spc_build.sh]: :+1:

File: hack/spc_build.sh (hunk 2 above)
Comment (TomSweeneyRedHat) [reply hack/spc_build.sh]: What I was commenting on here was on the left side of the github review screen, a number of tests were removed ala:

```
echo "buildah tag buildah-test-image..."
./buildah tag buildah-test-image localhost:5000/buildah-test-image
```
I'm not sure that they are covered by the conformance tests.  If you think we're hip, then I'm hip.

**Comment by TomSweeneyRedHat** [comment]
@lsm5 answering your question, I'm not 100% sure they're handled by the conformance tests.  Given that and the conformance test is likely to be turned off to start, I'd rather keep them unless you think it's unwise or completely duplicitous.   I'm fine with a TODO comment in there to remove them after the conformance tests go up fully if you think we're covered by that.

**Comment by lsm5** [comment]
@ypu @TomSweeneyRedHat I'd prefer everything be inside tests/conformance so people don't have to run `hack/spc_build.sh` separately and that hack/spc_build.sh just setup the fedora env and run `go test -v ./tests/conformance`

So, I'd prefer we move this hack/Dockerfile inside tests/conformance as well if needed.

WDYT?

**Comment by TomSweeneyRedHat** [comment]
@lsm5 I'm fine with that once the conformance tests are running properly.  Short term I'd like to leave them as is.  Noting the conformance tests are probably fine, the code needs some tweaking that's in progress.

**Comment by lsm5** [comment]
Let's resume this PR once tests are fixed in that case. Thanks @TomSweeneyRedHat

**Comment by lsm5** [comment]
Travis stuck again :\

**Comment by rhatdan** [comment]
@lsm5 I am fine with this, but @TomSweeneyRedHat has one last question that needs answering before we merge.

**Comment by lsm5** [comment]
@TomSweeneyRedHat so the latest travis run does run the test but doesn't block on it. Let me know if there are any other unresolved issues.

**Comment by lsm5** [comment]
@TomSweeneyRedHat also, `hack/Dockerfile` and related runs have been restored.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Merge this if you are happy with it.

**Comment by TomSweeneyRedHat** [comment]
LGTM @lsm5, thanks for getting this in and TY @ypu for putting together the conformance tests!

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot r+

**Comment by mrunalp** [comment]
:+1:

**Comment by lsm5** [comment]
thanks @TomSweeneyRedHat @rhatdan

**Comment by rhatdan** [comment]
Homu does not seem to be running.  Merging manually.

**Pull request merged**